### PR TITLE
Enhancement: Send user verification emails

### PR DESCRIPTION
**Pull request by weihesdlegend**
## Description
Send email verification when users sign up.

## Solution
* Create a mailer for SendGrid integration
* Create temp user to hold user data while waiting for user clicking on the verification link

## Testing
- [x] Integration testing on Heroku staging
- [ ] Added new unit tests

## Checks
- [x] Have you removed commented code?
- [x] Have you used gofmt to format your code?

### CODE REVIEW

File: iowrappers/users.go [hunk 1]
```diff
@@ -85,7 +88,7 @@ func (redisClient *RedisClient) FindUser(context context.Context, findUserBy Fin
 	return userView, nil
 }
 
-func (redisClient *RedisClient) CreateUser(context context.Context, userView user.View) (user.View, error) {
+func (redisClient *RedisClient) CreateUser(context context.Context, userView user.View, skipPasswordGeneration bool) (user.View, error) {
```
Comment (rwangsc18) [inline iowrappers/users.go:91]: curious why we need `skipPasswordGeneration` with user email verification? Is it trying to skip password encryption? IIUC, the password is from user input, and we encrypted it before storing it on Redis. Does email verification process add some additional steps?

**Review by rwangsc18 — COMMENTED** [review]
LGTM
left one comment, is the PR deployed somewhere, so we can test it?

File: iowrappers/users.go (hunk 1 above)
Comment (weihesdlegend) [reply iowrappers/users.go:91]: > LGTM left one comment, is the PR deployed somewhere, so we can test it?

Deployed on https://staging-vp.herokuapp.com/v1/

Comment (weihesdlegend) [reply iowrappers/users.go:91]: > 

Yes, the purpose is to skip password encryption. When a user signs up, we cannot store the original password but the encrypted one temporarily. So the user creation function has to deal with both cases.

**Review by sananand007 — APPROVED** [review]
lg :t-rex:

**Pull request merged**